### DISCUSSION TITLE
How to update the theme of a plotter (QtInteractor from pyvistaqt) that already exists

### DISCUSSION THREAD

**Discussion by ycli0536** [pyvista/pyvista#1472] [Q&A] [answered]
I find the global theme is a great update! Then I update my `pyvista` and `pyvistaqt` to use it. The examples I've seen in the doc ([Control Global and Local Plotting Themes](https://docs.pyvista.org/examples/02-plot/themes.html?highlight=theme)) are all plotting after setting/re-setting the theme. I want to know if it is possible to plot meshes before setting the theme, and then refresh it. I plan to use it to refresh the method `show_edges` for the existing plotter but I didn't find a feasible way. I may be stuck in a wrong understanding of this function.

The code I used looks like below:
```python
from UI_init.Ui_ViewWin import Ui_MainWindow
from pyvistaqt import QtInteractor, MainWindow
import pyvista as pv

class pyvistaWin(MainWindow, Ui_MainWindow):
    def __init__(self):
        super(pyvistaWin, self).__init__()
        self.setupUi(self)
        self.plotter = QtInteractor()
        self.verticalLayout.addWidget(self.plotter.interactor)
        self.action_Bounds.triggered.connect(self.show_edges)
    def show_edges(self):
        self.plotter.theme.show_edges = True
        self.plotter.update()
```
I tried to use `self.plotter.update()` to achieve refreshing the plotter but it doesn't look as I expected or understand. But after setting `self.plotter.theme.show_edges = True`, the next drawings (I will clear the plotter or remove some actors later) will meet expectations but not the previous one. I want to find a way to separate the `show_edges `from `add_mesh`. Do you have any ideas or suggestions to achieve it?

The packages in my environment and the report shown by `print(pv.Report())` are:

```
Backend TkAgg is interactive backend. Turning interactive mode on.
--------------------------------------------------------------------------------
  Date: Wed Jun 30 19:02:55 2021
                OS : Windows
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
       Environment : IPython
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon R7 200 Series
       GPU Version : 4.5.13596 Core Profile Context 20.10.35.02 27.20.1034.6
  Python 3.7.10 (default, Feb 26 2021, 13:06:18) [MSC v.1916 64 bit (AMD64)]
           pyvista : 0.31.3
               vtk : 9.0.1
             numpy : 1.19.2
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.6
            meshio : 4.3.11
        matplotlib : 3.2.2
         pyvistaqt : 0.5.0
             PyQt5 : 5.15.4
           IPython : 7.21.0
          colorcet : 1.0.0
             scipy : 1.6.1
        itkwidgets : 0.32.0
  Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for
  Intel(R) 64 architecture applications
--------------------------------------------------------------------------------
```

THANK YOU

**Comment by akaszynski** [comment, marked as answer]
> I tried to use self.plotter.update() to achieve refreshing the plotter but it doesn't look as I expected or understand. But after setting self.plotter.theme.show_edges = True

The way themes work is it sets the default parameters of the plotting methods, and as you found, only applies for new plots.  I think if you want to make all plots update to a theme, you'll have to introduce a complex callback that loops through all the existing meshes and updates their properties.  Best solution is to either set the global theme prior to plotting or to configure the plot theme prior to adding meshes.
